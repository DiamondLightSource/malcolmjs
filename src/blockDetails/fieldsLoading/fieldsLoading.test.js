import React from 'react';
import renderer from 'react-test-renderer';
import FieldsLoading from './fieldsLoading.component';

describe('Fields Loading', () => {
  it('renders correctly', () => {
    const fields = [
      { name: 'health', loading: true },
      { name: 'icon', loading: false },
      { name: 'parameters', loading: true },
    ];

    const tree = renderer
      .create(<FieldsLoading attributes={fields} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
