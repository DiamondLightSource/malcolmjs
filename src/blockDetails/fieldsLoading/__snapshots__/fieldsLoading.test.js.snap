// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields Loading renders correctly 1`] = `
<div>
  <p
    className="MuiTypography-root-5 MuiTypography-body1-14"
  >
    Loading attributes...
  </p>
  <div
    className="FieldsLoading-loadingRow-1"
  >
    <div
      className="MuiCircularProgress-root-31 MuiCircularProgress-colorSecondary-35 MuiCircularProgress-indeterminate-33 FieldsLoading-progress-3"
      role="progressbar"
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-36"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle-37 MuiCircularProgress-circleIndeterminate-39"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root-5 MuiTypography-body1-14 FieldsLoading-fieldName-2"
    >
      health
    </p>
  </div>
  <div
    className="FieldsLoading-loadingRow-1"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-40 FieldsLoading-success-4"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <p
      className="MuiTypography-root-5 MuiTypography-body1-14 FieldsLoading-fieldName-2"
    >
      icon
    </p>
  </div>
  <div
    className="FieldsLoading-loadingRow-1"
  >
    <div
      className="MuiCircularProgress-root-31 MuiCircularProgress-colorSecondary-35 MuiCircularProgress-indeterminate-33 FieldsLoading-progress-3"
      role="progressbar"
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-36"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle-37 MuiCircularProgress-circleIndeterminate-39"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root-5 MuiTypography-body1-14 FieldsLoading-fieldName-2"
    >
      parameters
    </p>
  </div>
</div>
`;
